x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -893,6 +896,8 @@ void runComputeOnRemoteCluster(
             transportService.getThreadPool().executor(ThreadPool.Names.SEARCH),
             computeListener.acquireAvoid()
         );
+        exchangeSink.setSource(exchangeSource);
+        exchangeSource.onFinishEarly(() -> executionInfo.markClusterAsPartial(clusterAlias));
```

Comment:
Does this run only on the coordinator?

Reply:
runComputeOnRemoteCluster actually happens on the remote cluster if I understand correctly.